elements/lisk-codec/src/codec.ts

```diff
@@ -113,6 +120,39 @@ export class Codec {
 		return (res as unknown) as T;
 	}
 
+	// For performance applications use decode() instead!
+	public decodeJSON<T>(schema: Schema, message: Buffer): T {
+		const decodedMessage: IteratableGenericObject = this.decode(
+			schema,
+			message,
+		);
+		const decodedMessageCopy = objectUtils.cloneDeep(decodedMessage);
+		decodedMessageCopy[Symbol.iterator] = iterator;
+
+		recursiveTypeCast(
+			'toJSON',
+			decodedMessageCopy,
+			(schema as unknown) as SchemaProps,
+			[],
+		);
+		return (decodedMessageCopy as unknown) as T;
+	}
+
+	// For performance applications use encode() instead!
+	public encodeJSON(schema: Schema, message: object): Buffer {
+		const messageCopy = objectUtils.cloneDeep(message);
+		(messageCopy as IteratableGenericObject)[Symbol.iterator] = iterator;
+
+		recursiveTypeCast(
+			'fromJSON',
+			messageCopy as IteratableGenericObject,
+			(schema as unknown) as SchemaProps,
```

Comment:
Can we use compiled schema here? If it's efficient for `encode` we may have its benefit here as well.

Reply:
Changing to compiled schema will mean a rewrite of this whole PR and was discussed with Shu for now not to do it. We can refactor in the future to use compiled schema.